token-erc-20/chaincode-go/chaincode/token_contract.go

```diff
@@ -723,7 +723,7 @@ func sub(b int, q int) (int, error) {
 	var diff int
 	diff = b - q
 
-	if (diff > b) == (b >= 0 && q >= 0) {
+	if (diff < b || diff < -q) == (b >= 0 && q <= 0) {
```

Comment:
The equation is getting confusing... could we simplify it by validating that both b and q are positive integers at the beginning of the function?  Same thing for add().

Reply:
That would weaken the generality of the function. And what is the significance of the overflow judgment in the previous source code.

Reply:
I am indeed proposing to weaken the generality of the function. Token balances and transfers should always utilize positive integers, right? So why is there a need to support negative integers in this chaincode? Having the function explicitly only support positive integers would make it more expressive and scoped to the intent wouldn't it?